packages/eas-build-job/src/customJob.ts

```diff
@@ -0,0 +1,135 @@
+import { z } from 'zod';
+
+const CommonCustomJobStepZ = z.object({
+  /**
+   * Unique identifier for the step.
+   *
+   * @example
+   * id: 'step1'
+   */
+  id: z.string().optional(),
+  /**
+   * Expression that determines whether the step should run.
+   * Based on the GitHub Actions job step `if` field (https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif).
+   *
+   * @example
+   * if: '${{ steps.step1.outputs.test == 'test' }}'
+   */
+  if: z.string().optional(),
+  /**
+   * The name of the step.
+   *
+   * @example
+   * name: 'Step 1'
+   */
+  name: z.string().optional(),
+  /**
+   * The working directory to run the step in.
+   *
+   * @example
+   * working-directory: './my-working-directory'
+   *
+   * @default '.' (the root of the repository)
+   */
+  working_directory: z.string().optional(),
+  /**
+   * Env variables override for the step.
+   *
+   * @example
+   * env:
+   *  MY_ENV_VAR: 'my-value'
+   *  ANOTHER_ENV_VAR: 'another-value'
+   */
+  env: z.record(z.string()).optional(),
+});
+
+export const CustomJobFunctionStepZ = CommonCustomJobStepZ.extend({
+  /**
+   * The custom EAS function to run as a step.
+   * It can be a function provided by EAS or a custom function defined by the user.
+   *
+   * @example
+   * uses: 'eas/build'
+   *
+   * @example
+   * uses: 'my-custom-function'
+   */
+  uses: z.string(),
+  /**
+   * The arguments to pass to the function.
+   *
+   * @example
+   * with:
+   *  arg1: 'value1'
```

Comment:
Indentation is wrong on the examples (levels are nested by 1 space, here and in other places)